apis/projectcontour/v1alpha1/contourconfig.go

```diff
@@ -84,6 +84,11 @@ type ContourConfigurationSpec struct {
 
 	// Tracing defines properties for exporting trace data to OpenTelemetry.
 	Tracing *TracingConfig `json:"tracing,omitempty"`
+
+	// useEndpointSlice configures contour to fetch endpoint data
+	// from k8s endpoint slices. defaults to false and reading endpoint
+	// data from the k8s endpoints.
+	UseEndpointSlice bool `json:"useEndpointSlice,omitempty"`
```

Comment:
I'd be interested to hear what the other maintainers think about adding a more generic `FeatureFlags` field instead.  It could be used for other things, and deprecating a value for it should be simpler than deprecating an entire field.

Reply:
I think thats a good idea, rather than polluting the config every time we want to add/deprecate a field.

Reply:
i made the change in https://github.com/projectcontour/contour/pull/5745/commits/1fd6eea12689e2de160d46537422f595af150d9b but will also wait for feedback from the other maintainers.

Reply:
I was actually thinking a string slice, rather than a concrete field per feature, so that as features come and go it doesn't require any changes to the type itself, just to the values that are accepted in the field. But, would like to get a second opinion before making this change.

Reply:
+1 for string slice, similar in spirit to the `--disable-feature` flag which is more generic than adding a specific flag per thing we want to disable

I guess that naming makes it sound more generic than it is, it's really for disabling reconciling some optional extended support CRDs, i dunno if we should repurpose the existing flag, have to think about it a bit

Reply:
Yeah, I was looking at that flag as well for possible use, but since its semantics are opt-out rather than opt-in, it seemed hard to use here (would you opt out of using Endpoints which would result in opting into EndpointSlices? Seems unintuitive). I also don't quite remember why we went with a flag vs. a config file/ContourConfig field for that, will have to look back. 

If we do just go ahead with a config file/ContourConfig field for opt-in features, then down the road we could always look at deprecating the `--disable-feature` flag and replacing with something more aligned with the opt-in feature flags.

Reply:
added a string slice in https://github.com/projectcontour/contour/pull/5745/commits/ee4aa23bff2d152d469d36cf1d6ccf7dd8ce4777

is this what y'all had in mind?

Reply:
Looks reasonable to me!